### PR TITLE
Fix the error when object key has special character

### PR DESCRIPTION
**Pull request by vincywindy**
set true to igonre the special char where path property is specified,

Fix https://github.com/aws/aws-sdk-net/issues/1943

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Comment by Edric-Li** [comment]
@vincywindy Has this Pull Request been merged and published?

**Comment by vincywindy** [comment]
> @vincywindy此拉取请求是否已合并并发布？

No

**Comment by Edric-Li** [comment]
Why isn't this merged?

**Comment by vincywindy** [comment]
> 为什么没有合并？

I don't know,Is there something I'm not doing？

**Comment by Edric-Li** [comment]
Is anyone still maintaining it? If there is no problem with this PR, can it be merged and released a new version as soon as possible?

**Comment by normj** [comment]
@Edric-Li and all sorry this issue hasn't gotten attention. As small as the PR looks it changes a very low level area and these encoding changes are always tricky to change so we don't fix one issue and cause another. We have it on our queue to get to but there are many competing priorties.

File: sdk/src/Core/Amazon.Util/AWSSDKUtils.cs [hunk 1]
```diff
@@ -366,7 +366,7 @@ public static string CanonicalizeResourcePath(Uri endpoint, string resourcePath,
                 }
             }
 
-            var canonicalizedResourcePath = AWSSDKUtils.JoinResourcePathSegments(encodedSegments, false);
+            var canonicalizedResourcePath = AWSSDKUtils.JoinResourcePathSegments(encodedSegments, true);
```
Comment (peterrsongg) [inline sdk/src/Core/Amazon.Util/AWSSDKUtils.cs:369]: Thanks for opening up a PR. The issue in question #1943 is complaining that we are encoding special characters twice. In the example listed (`"/%3A"`), `CanonicalizeResourcePath` returns `"/%25253A"` while the user is saying it should return `"%253A`. Changing this value from false to true doesn't fix the issue, it just completely ignores uri encoding altogether.

Whether or not this is a bug or by design requires more investigation but this PR needs tests and the solution does not fix the underlying issue so I'm going to go ahead and close the PR. I will leave the issue open as we investigate this further.

**Pull request closed without merging**